### PR TITLE
add 'make generate' script for Windows

### PR DESCRIPTION
**Pull request by apwojcik**
The PR from the series to enable MIGraphX on Windows.

### CODE REVIEW

**Comment by pfultz2** [comment]
This should be either a python or cmake script so it uses the same script for both windows and linux.

File: tools/generate.cmd [hunk 1]
```diff
@@ -0,0 +1,43 @@
+::
+:: The MIT License (MIT)
+::
+:: Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (causten) [inline tools/generate.cmd:4]: 2023

**Comment by IncinX** [comment]
> This should be either a python or cmake script so it uses the same script for both windows and linux.

I agree, creating a common Windows and Linux solution is preferred. Our initial set of UIF2 changes were explicitly instructed to not break Linux and creating a unifying script for this would impact it. Since this is not high priority, would you mind if we create a work item for it and address it after we upstream changes since this change works for Windows and has low impact for Linux?

**Comment by tperry-amd** [comment]
> This should be either a python or cmake script so it uses the same script for both windows and linux.

I agree, a unified script for Linux and Windows is preferred. Our mandate for these initial changes was to minimally impact Linux. Are you okay if we generate a work item to address this after all the Windows changes have made it upstream? The reasoning is that this change is blocking Windows and the work for a new unified script has a testing requirement for both Windows and Linux that will take some time and is lower priority compared to enabling the larger Windows effort.

**Comment by pfultz2** [comment]
> The reasoning is that this change is blocking Windows 

`make generate` isnt needed to build migraphx either on windows or linux. Its only needed when a developer changes one of the autogenerated files, which can be done on linux at PR time. So I dont see how this could be a blocker, especially if all the changes has been upstreamed already.

**Comment by tperry-amd** [comment]
> > The reasoning is that this change is blocking Windows
> 
> `make generate` isnt needed to build migraphx either on windows or linux. Its only needed when a developer changes one of the autogenerated files, which can be done on linux at PR time. So I dont see how this could be a blocker, especially if all the changes has been upstreamed already.

Fair point. It is still additional work that affects Linux more directly. Our prerogative for the initial changes was to minimally impact Linux and follow your patterns. The pattern here was to solve this problem with a shell script and we solved it with a shell script for Windows. We can change the pattern and create a unified script that works on both OS but it could potentially affect Linux which increases the testing work.

I think we should consider a change in prerogative. A modest increase in Linux risk is okay if we create a unified solution. We'll pull the change, move it to another branch, and hold onto the branch until we have time to do the unified solution.

**Pull request closed without merging**